force-app/main/fia/classes/KafkaFiaCaseHandler.cls

```diff
@@ -0,0 +1,78 @@
+public with sharing class KafkaFiaCaseHandler extends KafkaMessageProcessor implements IKafkaMessageConsumer{
+    Map<String, KafkaFiaCase_Model> keyToFiaCase = new Map<String, KafkaFiaCase_Model>();
+
+    public void processMessages(List<KafkaMessage__c> messages) {
+        for (KafkaMessage__c msg : messages) {
+            try {
+                KafkaFiaCase_Model fia = (KafkaFiaCase_Model) parse(KafkaFiaCase_Model.class, msg);
+                mapFiaCase(fia);
+                msg.CRM_Status__c = KafkaMessageService.STATUS_PROCESSED;
+            } catch (Exception e) {
+                msg.CRM_Status__c = KafkaMessageService.STATUS_ERROR;
+                msg.CRM_ErrorMessage__c = e.getMessage() + ' (' + e.getLineNumber() + ')';
+            }
+        }
+        convertAndUpsertFiaCases(messages);
+    }
+
+    private void mapFiaCase(KafkaFiaCase_Model fiaCase) {
+        String key = fiaCase.key;
+
+        if (!keyToFiaCase.containsKey(key)) {
+            keyToFiaCase.put(key, fiaCase);
+        } else if (isMostRecent(fiaCase.endretTidspunkt, keyToFiaCase.get(key).endretTidspunkt)) {
+            keyToFiaCase.put(key, fiaCase);
+        }
+    }
+
+    private void convertAndUpsertFiaCases(List<KafkaMessage__c> messages) {
+        List<Account> fiaCases = new List<Account>();
+        try {
+            fiaCases = getFiaCasesSObjects();
+            upsert fiaCases Account.INT_OrganizationNumber__c;
+        } catch (Exception e) {
+            String errorRef = logError(e.getMessage() + '/n' + e.getStackTraceString());
+            setMessageErrors(messages, errorRef);
+        }
+    }
+
+    private List<Account> getFiaCasesSObjects() {
+        Map<String, String> lastSaved = getLastSavedFiaCase(Account.class, keyToFiaCase.keyset());
+        List<Account> fiaCases = new List<Account>();
+
+        for (KafkaFiaCase_Model model : keyToFiaCase.values()) {
+ 
+            if (isNewOrChanged(model.key, model.hash, lastSaved)) {
+                Account fiaCase = convertToSObject(model);
```

Comment:
Denne linjen kan skape problemer når det kommer store datamengder. Inn i `convertToSObject()` metoden har du a SOQL spørring, pga limits burde man unngå SOQL i loops. For å fikse dette kan du hente alle virksomheter i en spørring og så lagre i en HashMap hvor nøkkel er orgnummer og verdi er kontoen.

Reply:
har benyttet den eksisterende map'en, accountMap, som du har laget i de delte hjelpe-klassene